### PR TITLE
Demote current position calculation out of loops

### PR DESCRIPTION
**Pull request by adenm-10**
…duce computation

### CODE REVIEW

**Comment by adenm-10** [comment]
For the calculations inside the loops of alignAxis, should I just get rid of them all together since we're setting position equal to 0 anyway or create a variable that keeps track of how many iterations of each loop occur and implement current position assignments after each loop? Also I feel like there should be more to my comments but that's the only benefit of moving the calculation I could think of.

**Comment by LSaber36** [comment]
For now, you can just assign the position at the end of alignAxis. Nothing can interrupt the code for that, except for ISRs. Even so, an ISR would resume the code in that function if interrupted, so the flow would be the same.

File: firmware/arduino/chessboard/motorDriver.ino [hunk 1]
```diff
@@ -242,11 +237,13 @@ uint8_t moveStraight(uint8_t motor[], int endCol, int endRow)
...
     digitalWrite(motor[STEP_PIN], HIGH);
-
-    // Updating current position per step
-    *currentMotorPos += eighthStepsPerPulse;
   }
 
+    // Updates current position of relevant motor
+    // Not incremented inside of loop to save runtime, and isn't needed as endstop collision
+    // is only exception to end position result, which will be dealt with by calling alignAxis
+    *currentMotorPos += (eighthStepsPerPulse * numSteps);
```
Comment (nashirj) [inline firmware/arduino/chessboard/motorDriver.ino:245]: Indentation is off here, need to decrease by two spaces. Also line 243 and 244 are a little confusing can u pls change it to one statement per line, something like:
```c
// Updated position is calculated outside of loop to reduce computation/runtime
// If motor collides with endstop, `alignAxis` is called to fix current position
```

File: firmware/arduino/chessboard/motorDriver.ino [hunk 2]
```diff
@@ -338,11 +335,13 @@ uint8_t moveDiagonal(int endCol, int endRow)
...
-    // Updates current position for both x and y
-    currPositionX += eighthStepsPerPulseX;
-    currPositionY += eighthStepsPerPulseY;
   }
 
+    // Updates current position for both X and Y motors
+    // Not incremented inside of loop to save runtime, and isn't needed as endstop collision
+    // is only exception to end position result, which will be dealt with by calling alignAxis
+    currPositionX += (eighthStepsPerPulseX * numStepsX);
+    currPositionY += (eighthStepsPerPulseY * numStepsX);
```
Comment (nashirj) [inline firmware/arduino/chessboard/motorDriver.ino:344]: Fix indentation here as well, and update comment as above.

**Review by nashirj — COMMENTED** [review]
Overall looks really good, just left you two comments.

**Pull request merged**